src/NzbDrone.Core/ImportLists/Sonarr/SonarrImport.cs

```diff
@@ -41,7 +41,8 @@ public override IList<ImportListItemInfo> Fetch()
                 {
                     if ((!Settings.ProfileIds.Any() || Settings.ProfileIds.Contains(item.QualityProfileId)) &&
                         (!Settings.LanguageProfileIds.Any() || Settings.LanguageProfileIds.Contains(item.LanguageProfileId)) &&
-                        (!Settings.TagIds.Any() || Settings.TagIds.Any(tagId => item.Tags.Any(itemTagId => itemTagId == tagId))))
+                        (!Settings.TagIds.Any() || Settings.TagIds.Any(tagId => item.Tags.Any(itemTagId => itemTagId == tagId))) &&
```

Comment:
I think this if statement is getting hairy enough that it could instead be broken up to return early (continue).

For example for profiles:

```
if (Settings.ProfileIds.Any() && !Settings.ProfileIds.Contains(item.QualityProfileId))
{
	continue;
}
```